provider/provider_python_test.go

```diff
@@ -70,6 +74,38 @@ func TestRegress2534(t *testing.T) {
 	execPulumi(t, ptest, workdir, "state", "unprotect", strings.ReplaceAll(targetGroupUrn, "::test", "::newtg"), "--yes")
 }
 
+func TestRegress4457(t *testing.T) {
+	ptest := pulumiTest(t, filepath.Join("test-programs", "regress-4457"))
+	upResult := ptest.Up()
+	autoGroupArn := upResult.Outputs["autoGroupArn"].Value.(string)
+	autoGroupUrn := upResult.Outputs["autoGroupUrn"].Value.(string)
+	autoGroupName := upResult.Outputs["autoGroupName"].Value.(string)
+	workspace := ptest.CurrentStack().Workspace()
+
+	t.Logf("Provisioned autoscaling group with arn=%s and urn=%s and name=%s", autoGroupArn, autoGroupUrn, autoGroupName)
+	workdir := workspace.WorkDir()
+	t.Logf("workdir = %s", workdir)
+	importResult := execPulumi(t, ptest, workdir, "import", "aws:autoscaling/group:Group", "newag", autoGroupName, "--yes")
+
+	t.Logf("Editing the program to add the code recommended by import")
```

Comment:
AFAIK we have no precedent of testing this, throwing in some automation, can generalize later if this is important.